nominal/nominal.py

```diff
@@ -1,628 +1,263 @@
-        self.relative = relative
-        self.relative_units = relative_units
-
-    def __get_headers(self, content_type: str = "json") -> dict:
-        TOKEN = kr.get_password("Nominal API", "python-client")
-        return {
-            "Authorization": "Bearer {}".format(TOKEN),
-            "Content-Type": "application/{0}".format(content_type),
-        }
-
-    def __upload_file(self, overwrite: bool) -> requests.Response:
-        """
-        Uploads dataframe to S3 as a file.
-
-        Returns:
-        Response object from the REST call.
-        """
-
-        if self.s3_path is not None and not overwrite:
-            print(
-                "\nThis Dataset is already uploaded to an S3 bucket:\n{0}\nTry [code]upload(overwrite = True)[/code] to overwrite it.".format(
-                    self.s3_path
-                )
-            )
-            return
-
-        # Create a default dataset name
-        if self.filename is None:
-            self.filename = default_filename("DATASET")
-
-        csv_file_buffer = io.BytesIO()
-        self.write_csv(csv_file_buffer, datetime_format=r"%Y-%m-%dT%H:%M:%S.%fZ")
-
-        # Get the size of the buffer in bytes
-        csv_file_buffer.seek(0, os.SEEK_END)
-        csv_buffer_size_bytes = csv_file_buffer.tell()
-        csv_file_buffer.seek(0)
-
-        print(
-            "\nUploading: [bold green]{0}[/bold green]\nto {1}\n = {2} bytes".format(
-                self.filename, get_base_url(), csv_buffer_size_bytes
-            )
-        )
-
-        # Make POST request to upload data file to S3
-        resp = requests.post(
-            url=ENDPOINTS["file_upload"].format(get_base_url(), self.filename),
-            data=csv_file_buffer.read(),
-            params={"sizeBytes": csv_buffer_size_bytes},
-            headers=self.__get_headers(content_type="octet-stream"),
-        )
 
-        if resp.status_code == 200:
-            self.s3_path = resp.text.strip('"')
-            print("\nUpload to S3 successful. S3 bucket:", self.s3_path)
-        elif resp.status_code == 401:
-            _auth_help_blurb()
-        else:
-            print("\n{0} error during upload to S3:\n".format(resp.status_code), resp.json())
+_DEFAULT_BASE_URL = "https://api.gov.nominal.io/api"
+_KEYRING_SERVICE = "nominal-python-client"
+_KEYRING_USER = "nominal-python-user"
 
-        return resp
+# global variable which `set_base_url()` is intended to modify
+_global_base_url = _DEFAULT_BASE_URL
 
-    def upload(self, overwrite: bool = False):
-        """
-        Registers Dataset in Nominal on Nominal platform.
 
-        Endpoint:
-        /ingest/v1/trigger-ingest-v2
+@cache
+def _get_or_create_connection(base_url: str, token: str) -> NominalClient:
+    return NominalClient.create(base_url, token)
 
-        Returns:
-        Response object from the REST call.
-        """
 
-        s3_upload_resp = self.__upload_file(overwrite)
+def set_base_url(base_url: str = _DEFAULT_BASE_URL) -> None:
```

Comment:
hmm now this is kinda wonky, `set_base_url` behaves differently than `set_token`.

I think I'm just going to roll in `~/.nominal.yml` now

Reply:
Okay, so now there's _only_ a `set_base_url()`. Setting a token should be done on the system / outside of the Python API to discourage checking in tokens. I included a CLI utility for creating/setting the config file.

```py
python -m nominal auth set-token
```

Reply:
I will eventually build out a more fully featured rust/clap CLI for this and other functionality in the near future, but this works for now